mirgecom/transport.py

```diff
@@ -316,6 +320,186 @@ def species_diffusivity(self, cv: ConservedVars,  # type: ignore[override]
+                         dv: GasDependentVars,
+                         eos: Optional[GasEOS] = None) -> DOFArray:
+        r"""Get the 2nd viscosity coefficent, $\lambda$.
+
+        In this transport model, the second coefficient of viscosity is defined as:
+
+        .. math::
+
+            \lambda = \left(\alpha - \frac{2}{3}\right)\mu
+
+        """
+        return (self._alpha - 2.0/3.0)*self.viscosity(cv, dv)
+
+    def thermal_conductivity(self, cv: ConservedVars,  # type: ignore[override]
+                             dv: GasDependentVars,
+                             eos: GasEOS) -> DOFArray:
+        r"""Get the gas thermal_conductivity, $\kappa$.
+
+        The thermal conductivity can be obtained from Pyrometheus using a
+        mixture averaged rule considering the species heat conductivities and
+        mole fractions:
+
+        .. math::
+
+            \kappa = \frac{1}{2} \left( \sum_{k=1}^{K} X_k \lambda_k +
+               \frac{1}{\sum_{k=1}^{K} \frac{X_k}{\lambda_k} }\right)
+
+        """
+        return self._factor*(self._pyro_mech.get_mixture_thermal_conductivity_mixavg(
+            dv.temperature, cv.species_mass_fractions,))
+
+    def species_diffusivity(self, cv: ConservedVars,  # type: ignore[override]
+                            dv: GasDependentVars,
+                            eos: GasEOS) -> DOFArray:
+        r"""Get the vector of species diffusivities, ${d}_{i}$.
+
+        The species diffusivities can be obtained directly from Pyrometheus using
+        a mixture averaged rule considering the species binary mass diffusivities
+        $d_{ij}$ and the mass fractions $Y_i$
+
+        .. math::
+
+            d_{i}^{(m)} = \frac{1 - Y_i}{\sum_{j\ne i} \frac{X_j}{d_{ij}}}
+
+        In regions with a single species, the above equation is ill-conditioned
+        and a constant diffusivity is used instead.
+
+        The user can prescribe an array with the Lewis number $Le$ for each species.
+        Then, it is used together with the mixture termal conductivity and the
+        heat capacity at constant pressure $C_p$ to yield the diffusivity.
+
+        .. math::
+
+            d_{i} = \frac{\kappa}{\rho \; Le \; C_p}
+
+        """
+        if self._lewis is not None:
+            return (self.thermal_conductivity(cv, dv, eos)/(
+                cv.mass*self._lewis*eos.heat_capacity_cp(cv, dv.temperature))
+            )
+
+        actx = cv.mass.array_context
+
+        diffusivity = self._pyro_mech.get_species_mass_diffusivities_mixavg(
+            dv.pressure, dv.temperature, cv.species_mass_fractions)
+
+        for i in range(0, self._pyro_mech.num_species):
+            # where "1-Yi < epsilon" means "Y_i -> 1.0"
+            diffusivity[i] = \
+                actx.np.where(
+                    actx.np.less(1.0 - cv.species_mass_fractions[i], self._epsilon),
+                    self._singular_diffusivity,
+                    diffusivity[i]
+                )
```

Comment:
Potential performance issue.   We'll just have to keep an eye on this when we start using it.